zenmap/zenmapGUI/ScanInterface.py

```diff
@@ -429,33 +430,46 @@ def start_scan_cb(self, widget=None):
         command = self.command_toolbar.command
         profile = self.toolbar.selected_profile
 
-        log.debug(">>> Start Scan:")
-        log.debug(">>> Target: '%s'" % target)
-        log.debug(">>> Profile: '%s'" % profile)
-        log.debug(">>> Command: '%s'" % command)
+        check = subprocess.Popen(["mktemp"], stdout=subprocess.PIPE).communicate()[0]
 
-        if target != '':
-            try:
-                self.toolbar.add_new_target(target)
-            except IOError, e:
-                # We failed to save target_list.txt; treat it as read-only.
-                # Probably it's owned by root and this is a normal user.
-                log.debug(">>> Error saving %s: %s" % (
-                    Path.target_list, str(e)))
-
-        if command == '':
+        if (not check):
             warn_dialog = HIGAlertDialog(
-                    message_format=_("Empty Nmap Command"),
-                    secondary_text=_("There is no command to execute. "
-                        "Maybe the selected/typed profile doesn't exist. "
-                        "Please check the profile name or type the nmap "
-                        "command you would like to execute."),
-                    type=gtk.MESSAGE_ERROR)
+                    message_format=_("Temp File Could not be created"),
+                    secondary_text=_("There is no usable temp directory. "
+                        "Please check file permissions and run nmap again"),
+                        type=gtk.MESSAGE_ERROR)
             warn_dialog.run()
             warn_dialog.destroy()
             return
 
-        self.execute_command(command, target, profile)
+        else:
```

Comment:
No need for `else` here, since the previous `if` block ended with with a `return`